### PR TITLE
Add 'define' conversation to Hackbot

### PR DESCRIPTION
**Pull request by maxwofford**

### CODE REVIEW

File: app/models/hackbot/conversations/define.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
+module Hackbot
```
Comment (paked) [inline app/models/hackbot/conversations/define.rb]: Looks like we're starting to repeat some of this code...

**Comment by paked** [comment]
Before we merge this, I'd like to have a general discussion on how we're currently modelling these sorts of one-off-commands.

Current problems:

- This will spew errors on events where `event[:text].nil?`
- We're repeating the `event_to_query` code. I actually think there is a much nicer way we could implement it too.

```
def event_to_query(command, event)
  event[:text].sub(/.*#{command}/, '')
end
```

Even so, it'd be nice to have a `SimpleCommand` conversation which handled these things for us:

```ruby
class Define < Hackbot::Conversations::SimpleCommand
  set_command 'define'

  def do(event)
     query = get_query
     
    if ...handle empty case ... then ... end
    
   msg_channel UrbanDictionaryClient.define query
  end
end
```

**Comment by MaxWofford** [comment]
@paked, starting to question if we really want this in the Slack after the messages it was posting in #bot-spam. What are your thoughts on us still going forward with this?

**Comment by paked** [comment]
@MaxWofford: Unless we can find a way to make it PG, I'm not really sure if it's a good idea.

@zachlatta: Thoughts? context: https://hackclub.slack.com/archives/bot-spam/p1487897908759868

**Comment by zachlatta** [comment]
Yeah, I say either drop the feature or implement it using a regular dictionary API.

One alternative would be to introduce a list of "naughty words" and check what Urban Dictionary returns against it, but that seems a little overkill (and I imagine there'd be false positives & negatives).

**Comment by MaxWofford** [comment]
@zachlatta @paked: I'm going to close this for now. There are some good ways to implement this that can ensure it stays "PG", but this should really be done through a single-message command interface for hackbot. I'll create an adapter that can make define requests once we have that up.

**Pull request closed without merging**